internal/provider/data_source_helpers.go

```diff
@@ -4,53 +4,56 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-// datasourceSchemaFromResourceSchema is a recursive func that
-// converts an existing Resource schema to a Datasource schema.
-// All schema elements are copied, but certain attributes are ignored or changed:
-// - all attributes have Computed = true.
-// - all attributes have ForceNew, Required = false.
-// - Validation funcs and attributes (e.g. MaxItems) are not copied.
-func datasourceSchemaFromResourceSchema(rs map[string]*schema.Schema) map[string]*schema.Schema {
-	ds := make(map[string]*schema.Schema, len(rs))
-	for k, v := range rs {
-		dv := &schema.Schema{
+// dataSourceSchemaFromResourceSchema is a recursive func that converts an existing
+// Resource schema to a Datasource schema.
+//
+// All schema properties are copied, but some are ignored or changed:
+// - All fields have Computed = true.
+// - All fields have ForceNew, Required = false.
+// - Validation and attributes (e.g. MaxItems) are not copied.
+func dataSourceSchemaFromResourceSchema(resourceSchema map[string]*schema.Schema) map[string]*schema.Schema {
```

Comment:
Nested objects don't appear to have the descriptions passed through on the generated docs so it was believed that there was a flaw within this func that's why the refactor. I added additional test cases as well to ensure the description gets correctly retained. However there was no issue with this func but with the description generator for nested objects for data sources within the tfplugindocs.